homeassistant/components/drop_connect/sensor.py

```diff
@@ -0,0 +1,272 @@
+"""Support for DROP sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CONF_COORDINATOR,
+    CONF_DEVICE_TYPE,
+    DEV_FILTER,
+    DEV_HUB,
+    DEV_LEAK_DETECTOR,
+    DEV_PROTECTION_VALVE,
+    DEV_PUMP_CONTROLLER,
+    DEV_RO_FILTER,
+    DEV_SOFTENER,
+    DOMAIN,
+)
+from .coordinator import DROPDeviceDataUpdateCoordinator
+from .entity import DROPEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+FLOW_ICON = "mdi:shower-head"
+GAUGE_ICON = "mdi:gauge"
+TDS_ICON = "mdi:water-opacity"
+
+
+@dataclass(kw_only=True)
+class DROPSensorEntityDescription(SensorEntityDescription):
+    """Describes DROP sensor entity."""
+
+    value_fn: Callable[[DROPDeviceDataUpdateCoordinator], float | int | None]
+
+
+SENSORS: dict[str, DROPSensorEntityDescription] = {
+    "current_flow_rate": DROPSensorEntityDescription(
```

Comment:
Can these be constants?

Reply:
For me this feels like duplication to set the key of the dictionary, and the same key for the description. Perhaps you can just provide a list of descriptions with a key? 

You can then loop through this list during init of the sensors.

Reply:
Or also nice, if you also own the library. Provide an enum with all options from your library.

```
@unique
class SensorType(StrEnum):
    CURRENT_FLOW_RATE = "current_flow_rate"
```